File: backend/src/CorporateTravel.Application/Services/PasswordRequirementsService.cs
using CorporateTravel.Application.Interfaces;
using System.Text.RegularExpressions;

namespace CorporateTravel.Application.Services;

public class PasswordRequirementsService : IPasswordRequirementsService
{
    public PasswordRequirements GetPasswordRequirements()
    {
        return new PasswordRequirements
        {
            MinimumLength = 8,
            RequireDigit = true,
            RequireLowercase = true,
            RequireUppercase = true,
            RequireNonAlphanumeric = true,
            Description = "A senha deve ter pelo menos 8 caracteres, incluindo pelo menos uma letra maiúscula, uma minúscula, um dígito e um caractere especial."
        };
    }

    public bool ValidatePassword(string password, out List<string> errors)
    {
        errors = new List<string>();

        if (string.IsNullOrEmpty(password))
        {
            errors.Add("A senha é obrigatória.");
            return false;
        }

        if (password.Length < 8)
        {
            errors.Add("A senha deve ter pelo menos 8 caracteres.");
        }

        if (!Regex.IsMatch(password, @"\d"))
        {
            errors.Add("A senha deve conter pelo menos um dígito.");
        }

        if (!Regex.IsMatch(password, @"[a-z]"))
        {
            errors.Add("A senha deve conter pelo menos uma letra minúscula.");
        }

        if (!Regex.IsMatch(password, @"[A-Z]"))
        {
            errors.Add("A senha deve conter pelo menos uma letra maiúscula.");
        }

        if (!Regex.IsMatch(password, @"[^a-zA-Z0-9]"))
        {
            errors.Add("A senha deve conter pelo menos um caractere especial.");
        }

        return errors.Count == 0;
    }
} 

File: frontend/src/app/core/password-requirements.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface PasswordRequirements {
  minimumLength: number;
  requireDigit: boolean;
  requireLowercase: boolean;
  requireUppercase: boolean;
  requireNonAlphanumeric: boolean;
  description: string;
}

@Injectable({
  providedIn: 'root'
})
export class PasswordRequirementsService {
  private requirements: PasswordRequirements | null = null;

  constructor(private http: HttpClient) {}

  getPasswordRequirements(): Observable<PasswordRequirements> {
    return this.http.get<PasswordRequirements>(`${environment.apiUrl}/auth/password-requirements`);
  }

  validatePassword(password: string): { isValid: boolean; errors: string[] } {
    const errors: string[] = [];

    if (!password) {
      errors.push('A senha é obrigatória.');
      return { isValid: false, errors };
    }

    if (password.length < 8) {
      errors.push('A senha deve ter pelo menos 8 caracteres.');
    }

    if (!/\d/.test(password)) {
      errors.push('A senha deve conter pelo menos um dígito.');
    }

    if (!/[a-z]/.test(password)) {
      errors.push('A senha deve conter pelo menos uma letra minúscula.');
    }

    if (!/[A-Z]/.test(password)) {
      errors.push('A senha deve conter pelo menos uma letra maiúscula.');
    }

    if (!/[^a-zA-Z0-9]/.test(password)) {
      errors.push('A senha deve conter pelo menos um caractere especial.');
    }

    return { isValid: errors.length === 0, errors };
  }

  getPasswordStrength(password: string): { strength: 'weak' | 'medium' | 'strong'; percentage: number } {
    let score = 0;
    
    if (password.length >= 8) score += 20;
    if (password.length >= 12) score += 10;
    if (/\d/.test(password)) score += 20;
    if (/[a-z]/.test(password)) score += 20;
    if (/[A-Z]/.test(password)) score += 20;
    if (/[^a-zA-Z0-9]/.test(password)) score += 10;

    if (score < 40) return { strength: 'weak', percentage: score };
    if (score < 80) return { strength: 'medium', percentage: score };
    return { strength: 'strong', percentage: score };
  }
} 